sagemaker_processing/feature_transformation_with_sagemaker_processing_Dask/feature_transformation_with_sagemaker_processing_dask.ipynb

```diff
@@ -0,0 +1,490 @@
+    "\n",
+    "1. [Objective](#Objective:-predict-the-age-of-an-Abalone-from-its-physical-measurement)\n",
+    "1. [Setup](#Setup)\n",
+    "1. [Using Amazon SageMaker Processing to execute a Dask Job](#Using-Amazon-SageMaker-Processing-to-execute-a-Dask-Job)\n",
+    "  1. [Downloading dataset and uploading to S3](#Downloading-dataset-and-uploading-to-S3)\n",
+    "  1. [Build a Dask container for running the preprocessing job](#Build-a-Dask-container-for-running-the-preprocessing-job)\n",
+    "  1. [Run the preprocessing job using Amazon SageMaker Processing](#Run-the-preprocessing-job-using-Amazon-SageMaker-Processing)\n",
+    "    1. [Inspect the preprocessed dataset](#Inspect-the-preprocessed-dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "Let's start by specifying:\n",
+    "* The S3 bucket and prefixes that you use for training and model data. Use the default bucket specified by the Amazon SageMaker session.\n",
+    "* The IAM role ARN used to give processing and training access to the dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from time import gmtime, strftime\n",
+    "import sagemaker\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "role = sagemaker.get_execution_role()\n",
+    "bucket = sagemaker_session.default_bucket()\n",
+    "timestamp_prefix = strftime(\"%Y-%m-%d-%H-%M-%S\", gmtime())\n",
+    "\n",
+    "prefix = \"sagemaker/dask-preprocess-demo\"\n",
+    "input_prefix = prefix + \"/input/raw/census\"\n",
+    "input_preprocessed_prefix = prefix + \"/input/preprocessed/census\"\n",
+    "model_prefix = prefix + \"/model\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using Amazon SageMaker Processing to execute a Dask job"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Downloading dataset and uploading to Amazon Simple Storage Service (Amazon S3)\n",
+    "\n",
+    "The dataset used here is the Census-Income KDD Dataset. You select features from this dataset, clean the data, and turn the data into features that the training algorithm can use to train a binary classification model, and split the data into train and test sets. The task is to predict whether rows representing census responders have an income greater than $50,000, or less than $50,000. The dataset is heavily class imbalanced, with most records being labeled as earning less than $50,000. After training a logistic regression model, you evaluate the model against a hold-out test dataset, and save the classification evaluation metrics, including precision, recall, and F1 score for each label, and accuracy and ROC AUC for the model. In this notebook, you download from that bucket and upload to your own bucket so that Amazon SageMaker can access the dataset."
```

Comment:
I don't think this example does **all** of these things (just the pre-processing steps)

Reply:
All done and fixed